fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/mixin/resources/MixinMinecraftGame.java

```diff
@@ -16,66 +16,48 @@
 
 package net.fabricmc.fabric.mixin.resources;
 
-import com.google.common.collect.Lists;
-import net.fabricmc.fabric.impl.resources.ModResourcePackUtil;
+import net.fabricmc.fabric.api.event.resource.PackProvisionCallback;
+import net.fabricmc.fabric.impl.resources.CustomInjectionResourcePackProfile;
+import net.fabricmc.fabric.impl.resources.ModResourcePackProvider;
 import net.minecraft.client.MinecraftClient;
-import net.minecraft.client.resource.DefaultClientResourcePack;
-import net.minecraft.resource.ReloadableResourceManager;
+import net.minecraft.client.resource.ClientResourcePackContainer;
 import net.minecraft.resource.ResourcePack;
+import net.minecraft.resource.ResourcePackContainer;
+import net.minecraft.resource.ResourcePackContainerManager;
+import net.minecraft.resource.ResourcePackCreator;
 import net.minecraft.resource.ResourceType;
 import org.spongepowered.asm.mixin.Mixin;
-import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
-import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.Redirect;
-import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
-import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+import org.spongepowered.asm.mixin.injection.Slice;
 
-import java.util.List;
-import java.util.concurrent.CompletableFuture;
-import java.util.stream.Collector;
-import java.util.stream.Collectors;
+import java.util.function.Function;
 import java.util.stream.Stream;
 
 @Mixin(MinecraftClient.class)
 public class MixinMinecraftGame {
-	@Shadow
-	private ReloadableResourceManager resourceManager;
 
-	private void fabric_modifyResourcePackList(List<ResourcePack> list) {
-		List<ResourcePack> oldList = Lists.newArrayList(list);
-		list.clear();
-
-		boolean appended = false;
-		for (int i = 0; i < oldList.size(); i++) {
-			ResourcePack pack = oldList.get(i);
-			list.add(pack);
-
-			if (pack instanceof DefaultClientResourcePack) {
-				ModResourcePackUtil.appendModResourcePacks(list, ResourceType.CLIENT_RESOURCES);
-				appended = true;
-			}
-		}
-
-		if (!appended) {
-			StringBuilder builder = new StringBuilder("Fabric could not find resource pack injection location!");
-			for (ResourcePack rp : oldList) {
-				builder.append("\n - ").append(rp.getName()).append(" (").append(rp.getClass().getName()).append(")");
-			}
-			throw new RuntimeException(builder.toString());
-		}
+	@Redirect(method = "<init>", at = @At(value = "INVOKE", target = "Lnet/minecraft/resource/ResourcePackContainerManager;addCreator(Lnet/minecraft/resource/ResourcePackCreator;)V"),
+		slice = @Slice(from = @At(value = "INVOKE", target = "Lnet/minecraft/resource/FileResourcePackCreator;<init>(Ljava/io/File;)V"),
+			to = @At(value = "FIELD", target = "Lnet/minecraft/client/MinecraftClient;netProxy:Ljava/net/Proxy;", ordinal = 0)))
```

Comment:
Can you inject after the `addCreator` call instead of redirecting it?

Reply:
there are 2 addCreator calls. i guess redirecting is safer?

Reply:
Fixed